src/lib.rs

```diff
@@ -321,4 +322,42 @@ impl Contract {
 
         notify::notify_edit(id, post_author);
     }
+
+    pub fn add_community(&mut self, slug: String, mut community: Community) {
+        if let Some(_) = self.communities.get(&slug) {
+            panic!("Community already exists");
+        }
+
+        community.validate();
+        community.set_default_admin();
+        self.communities.insert(&slug, &community);
+    }
+
+    pub fn edit_community(&mut self, slug: String, mut community: Community) {
+        let community_old = self.get_community(slug.clone());
+        if !community_old.admins.contains(&env::predecessor_account_id()) {
+            panic!("Only community admins can edit community");
+        }
+
+        community.validate();
+        community.set_default_admin();
+        self.communities.insert(&slug, &community);
+    }
+
+    pub fn get_all_communities(&self) -> Vec<CommunityCard> {
+        near_sdk::log!("get_all_communities");
+        self.communities
+            .iter()
+            .map(|(slug, community)| CommunityCard {
+                slug,
+                name: community.name,
+                description: community.description,
+                image: community.image,
+            })
+            .collect()
+    }
+
+    pub fn get_community(&self, slug: String) -> Community {
+        self.communities.get(&slug).expect(&format!("Community {} doesn't exist", slug))
```

Comment:
I did not notice that we crash in `get_post`, but I would consider using `Option<Community>` return value instead of crashing the execution of a view function (and update `get_post` implementation to match it)

Reply:
I will update it Option<Community>. Change `get_post` would be a breaking change to UI, which would be better to address separately.